material/lessons/switzerland-dual-use/recipe_switzerland-dual-use.Rmd

```diff
@@ -286,59 +291,65 @@ to each country.
 
-```{r, fig.height=12, fig.width=7}
+```{r sort_by_number_of_exports}
+# This code groups elic_2016_4 by the variable country ('Land'), then uses summarise() to find the number of rows (i.e. exports) that went to each. 
 n_land <- summarise(group_by(elic_2016_4, Land), n = NROW(Land))
+
+# order() then sorts the data set in terms of exports, i.e. n, in ascending order
 ord <- order(n_land$n)
 str(ord)
 ordered_land <- n_land$Land[ord]
 str(ordered_land)
+# Here we convert the column 'Land' to the factor variable type, and set the order of the levels (i.e. values) using the ordered vector of country names 'ordered_land'.  This makes the data print or plot in this order when the factor variable is used.
 elic_2016_4$Land <- factor(elic_2016_4$Land, levels = ordered_land)
 str(elic_2016_4$Land)
 
+```
+The next plot makes a horizontal bar chart.  When specified in this way, the length of the bars is determined by the number of times that a value shows up in the data.  In this case, that means the number of times a country occurs in the column `Land` in data set `elic_2016_4`.
+
+```{r, fig.height=12, fig.width=7}
 ggplot(elic_2016_4, aes(x = Land)) + 
   geom_bar() +
   coord_flip()
 ```
 
 
-And now one, which shows how much worth the goods were that 
-were exported to each country. Again we want to have that in a specific order.
+The following plot shows the worth of the goods that were exported to each country. Again we want to have the countries in a specific order, but this time the value we use to sort the data is the sum of the monitary value (the variable `Wert`) of exports to each country.  So we re-order the countries based on the summed export value, and assign this order in the `levels =` specification.
 
-```{r, fig.height=12, fig.width=7}
+```{r} 
 wert_land <- summarise(group_by(elic_2016_4, Land), sum = sum(Wert))
 ordered_land <- wert_land$Land[order(wert_land$sum)]
 elic_2016_4$Land <- factor(elic_2016_4$Land, levels = ordered_land)
+```
+
+The plot is similar, but by writing `stat = "identity"`, the total value of `Wert` is plotted for each country, and these are displayed in the order we just specified above.
 
+```{r, fig.height=12, fig.width=7}
 ggplot(elic_2016_4, aes(x = Land, y = Wert, fill = Typ)) + 
   geom_bar(stat = "identity") +
   coord_flip()
 ```
 
-
 But wait, why are there countries listed, that do not seem to have gotten anything
 from switzerland? Let's look at those:
 
-```{r, fig.height=2, fig.width=7}
-p <- ggplot(elic_2016_4, aes(x = Wert)) + stat_ecdf()
-p
-
-p + coord_cartesian(xlim = c(0, 10^5))
+```{r}
+library('knitr') # kable() in this package will make nice tables
 
 novalue  <- filter(.data = elic_2016_4, Wert == 0)
-str(novalue)
+kable(summarize(group_by(.data=novalue,Land,Typ),n=n()),format = "html",caption = "Number Exports without Declared Value")
```

Comment:
This is maybe a bit too complicated, don't you think? Maybe split it up into more steps?

Reply:
I will simplfy this, but there have been additional modifications since my pull request, and I'd like to pull the entire repo before making the changes.  If I do that now, without you having done a merge, I think I will lose the changes I made.